response.go

```diff
@@ -13,32 +13,55 @@ const (
 // A ResponseWriter interface is used to respond to an event
 type ResponseWriter interface {
 	Reply(text string)
+	PostMessage(channel, message string, attachments *[]slack.Attachment)
 	ReportError(err error)
 	Typing()
+	RTM() *slack.RTM
+	Client() *slack.Client
 }
 
 // NewResponse creates a new response structure
-func NewResponse(channel string, RTM *slack.RTM) *Response {
-	return &Response{channel: channel, RTM: RTM}
+func NewResponse(channel string, Client *slack.Client, RTM *slack.RTM) *Response {
+	return &Response{channel: channel, client: Client, rtm: RTM}
 }
 
 // Response contains the channel and Real Time Messaging library
 type Response struct {
 	channel string
-	RTM     *slack.RTM
+	client  *slack.Client
+	rtm     *slack.RTM
 }
 
 // Reply send a message back to the channel where we received the event from
 func (r *Response) Reply(text string) {
-	r.RTM.SendMessage(r.RTM.NewOutgoingMessage(text, r.channel))
+	r.rtm.SendMessage(r.rtm.NewOutgoingMessage(text, r.channel))
 }
 
 // ReportError sends back a formatted error message to the channel where we received the event from
 func (r *Response) ReportError(err error) {
-	r.RTM.SendMessage(r.RTM.NewOutgoingMessage(fmt.Sprintf(errorFormat, err.Error()), r.channel))
+	r.rtm.SendMessage(r.rtm.NewOutgoingMessage(fmt.Sprintf(errorFormat, err.Error()), r.channel))
 }
 
 // Typing send a typing indicator
 func (r *Response) Typing() {
-	r.RTM.SendMessage(r.RTM.NewTypingMessage(r.channel))
+	r.rtm.SendMessage(r.rtm.NewTypingMessage(r.channel))
+}
+
+// PostMessage send a attachments to the current channel with a message
+func (r *Response) PostMessage(channel, message string, attachments *[]slack.Attachment) {
+	params := slack.PostMessageParameters{}
+	params.User = r.rtm.GetInfo().User.ID
+	params.AsUser = true
+	params.Attachments = *attachments
+	r.rtm.PostMessage(channel, message, params)
+}
```

Comment:
I like this addition. However, I feel it should be consistent with “Typing” and “Reply” in that the “channel” should be hidden and not required to be provided since most calling this method will intend on posting back to where the bot was called from.

And since there is already a “Reply”, that publishes a message, perhaps an optional “WithAttachments” parameter would be preferred. This way users can call “Reply(‘Hello’)” or “Reply(‘Attachments!’, WithAttachments(attachments))”

The RTM is exported, to allow users to do whatever advanced things they wish to do such as publishing to different channels and/or any other things such as shown in the examples folder

Reply:
Will do! 

Yes and no, if you put everything in the main is but not if you are using the approch that each command have its own file

Reply:
done, @shomali11  let me know if everything is ok